ksql-execution/src/main/java/io/confluent/ksql/execution/plan/StreamStreamJoin.java

```diff
@@ -25,21 +26,30 @@
 
   private final ExecutionStepProperties properties;
   private final JoinType joinType;
-  private final Formats formats;
+  private final Formats leftFormats;
+  private final Formats rightFormats;
   private final ExecutionStep<S> left;
   private final ExecutionStep<S> right;
+  private final Duration before;
+  private final Duration after;
 
   public StreamStreamJoin(
       final ExecutionStepProperties properties,
       final JoinType joinType,
-      final Formats formats,
+      final Formats leftFormats,
+      final Formats rightFormats,
       final ExecutionStep<S> left,
-      final ExecutionStep<S> right) {
+      final ExecutionStep<S> right,
+      final Duration before,
```

Comment:
Why not pass `JoinWindows` here?

Reply:
It's from Kafka Streams. Feels weird to put it into the step pojos.